corehq/apps/locations/permissions.py

```diff
@@ -319,6 +319,17 @@ def user_can_access_any_location_id(domain, user, location_ids):
             .exists())
 
 
+def user_can_access_other_user(domain, user, other_user):
+    if user.has_permission(domain, 'access_all_locations'):
+        return True
+
+    accessible_location_ids = (SQLLocation.objects
+        .accessible_to_user(domain, user)
+        .values_list('id', flat=True))
+
+    return other_user.sql_locations.filter(id__in=accessible_location_ids).count() > 0
```

Comment:
Is there ever a time we'd want to use `sql_locations`? If not, I think we should make that property private (out of scope of this PR, but is obviously confusing).